DashAI/back/core/schema_fields/component_field.py

```diff
@@ -0,0 +1,41 @@
+from beartype import beartype
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
+
+
+class ComponentType(BaseModel):
+    component: str
+    params: dict
+
+
+@beartype
+def component_field(
+    description: str,
+    default: str,
+    parent: str,
+):
+    """Function to create a custom pydantic-like type to support components.
+
+    Parameters
+    ----------
+    description: str
+        Description of the field.
+    default: str
+        The default value to show to the user.
+    parent: str
+        The name of the parent class of the component.
+        Is used to select the components to show to the user.
+
+    Returns
+    -------
+    type[ComponentType]
+        A pydantic-like type to represent the component.
+    """
+    return Annotated[
+        ComponentType,
+        Field(
+            description=description,
+            default=default,
```

Comment:
Are we leaving the default option then? We could also merge this to the project branch and then fix the "default problem". I think something like having `placeholder=placeholder` would work better than a default, because that would not remove it from the `required` field

Reply:
No, the default will be changed into placeholder, but in another PR. There is a Task for that in Notion https://www.notion.so/dash-ai/Change-default-to-placeholder-a22d7e73a10645c8a0a25205ad845a29?pvs=4

Reply:
Yes., that's the idea. I create [this](https://www.notion.so/dash-ai/Change-default-to-placeholder-a22d7e73a10645c8a0a25205ad845a29?pvs=4) Notion Task to solve it later.